[RLlib] Better behavior if user does not specify stopping condition in RLLib CLI

ArturNiederfahrenhorst opened the pull request:
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

We have lots of examples inside RLlib of running ray CLI similar to `rllib train --env ... --config ...` that lack a stopping condition.
They error out with the following message.

<img width="705" alt="Screenshot 2022-12-13 at 23 06 18" src="https://user-images.githubusercontent.com/9356806/207454214-f86af816-5406-4dc8-b5e7-cec9e3aafdd0.png">

We should not error out, since a workflow to be expected from us is to quickly start an experiment from CLI, observe it and cancel it manually without caring for stopping condidations.

## Related issue number

https://discuss.ray.io/t/problem-with-rllib-configuration-by-command-line/8624

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

maxpumperla commented on rllib/common.py:
```diff
@@ -216,7 +218,7 @@ class CLIArguments:
     # Train arguments
     # __cli_train_start__
-    Stop = typer.Option(None, "--stop", "-s", help=get_help("stop"))
+    Stop = typer.Option(None, *STOP_ARG_NAMES, help=get_help("stop"))
```
@ArturNiederfahrenhorst hmm, I think you can fix all of this by making the default arg `"{}"` instead of `None`

maxpumperla reviewed (changes requested):
great, thanks for spotting this @ArturNiederfahrenhorst! I think we can simplify

ArturNiederfahrenhorst commented:
@maxpumperla I have also removed the warning I introduced about stop being empty to simplify this.
Do you think that's a good idea or would you like it back?

maxpumperla commented:
@ArturNiederfahrenhorst I think we can add this to the typer help of the stop command, if not already there

ArturNiederfahrenhorst commented:
@maxpumperla Info on how to use stop is already in the help string. I'll leave it there and not warn users when it's missing.
